src/encoded/static/components/cart/cart.js

```diff
@@ -1571,9 +1571,17 @@ const processFilesAnalyses = (files, analyses) => {
 };
 
 
+/**
+ * Sort an array of files by their statuses according to their order in
+ * pseudoDefaultFileStatusOrder. Anything not in pseudoDefaultFileStatusOrder gets sorted randomly
+ * at the end.
+ */
 const pseudoDefaultFileStatusOrder = ['released', 'archived', 'in progress'];
 const sortPseudoDefaultFilesByStatus = (files) => (
-    _(files).sortBy((file) => pseudoDefaultFileStatusOrder.findIndex((status) => status === file.status))
+    _(files).sortBy((file) => {
+        const foundIndex = pseudoDefaultFileStatusOrder.indexOf(file.status);
```

Comment:
In Javascript sort in evergreen browsers, 0 has a special meaning (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort). The length is +1 larger than the range of values. There may be some odd corner cases that may make this wanky. But I do not think it is a big deal.